packages/core/src/package/componentFilter/EntitlementVersionFilter.ts

```diff
@@ -0,0 +1,78 @@
+import SFPOrg from '../../org/SFPOrg';
+import QueryHelper from '../../queryHelper/QueryHelper';
+import SFPLogger, { Logger, LoggerLevel } from '../../logger/SFPLogger';
+import { DeploymentFilter } from './DeploymentFilter';
+import * as fs from 'fs-extra';
+const { XMLBuilder } = require('fast-xml-parser');
+
+const EXISTING_SLAPPROCESS_QUERY = `SELECT Name, NameNorm,VersionNumber, VersionMaster FROM SlaProcess ORDER BY VersionNumber DESC`;
+
+export default class EntitlementVersionFilter implements DeploymentFilter {
+    public async apply(org: SFPOrg, componentSet: ComponentSet,logger:Logger): Promise<ComponentSet> {
+        //Only do if entitlment exits in the package
+        SFPLogger.log(`Filtering Entitlement Process....`,LoggerLevel.INFO,logger);
+        let sourceComponents = componentSet.getSourceComponents().toArray();
+        let isEntitlementFound: boolean = false;
+        for (const sourceComponent of sourceComponents) {
+            if (sourceComponent.type.name === registry.types.entitlementprocess.name) {
+                isEntitlementFound = true;
+                break;
+            }
+        }
+        if (!isEntitlementFound) return componentSet;
+
+        //Fetch Entitlements currently in the org
+        let slaProcessesInOrg = await QueryHelper.query<SlaProcess>(
+            EXISTING_SLAPPROCESS_QUERY,
+            org.getConnection(),
+            false
+        );
+
+        let modifiedComponentSet = new ComponentSet();
+
+        //Compare version numbers in the org vs version in the component set
+        //Remove if the version numbers match
+        for (const sourceComponent of sourceComponents) {
+            if (sourceComponent.type.name === registry.types.entitlementprocess.name) {
+                let slaProcessLocal = sourceComponent.parseXmlSync();
+
+                let slaProcessMatchedByName: SlaProcess = slaProcessesInOrg.find(
+                    (element: SlaProcess) => element.Name == slaProcessLocal['EntitlementProcess']['name'] && element.VersionNumber!=null
+                );
+                if (!slaProcessMatchedByName) {
+                    //Doesnt exist, deploy
+                    modifiedComponentSet.add(sourceComponent);
+                }
+                else if (slaProcessLocal['EntitlementProcess']['versionNumber'] > slaProcessMatchedByName.VersionNumber) {
+                    //This is a deployment candidate
+                    //Modify versionMaster tag to match in the org
+                    slaProcessLocal['EntitlementProcess']['versionMaster'] = slaProcessMatchedByName.VersionMaster;
+                    let builder = new XMLBuilder({ format: true, ignoreAttributes: false, attributeNamePrefix: '@_' });
+                    let xmlContent = builder.build(slaProcessLocal);
+                    fs.writeFileSync(sourceComponent.xml, xmlContent);
+                    modifiedComponentSet.add(sourceComponent);
```

Comment:
I think we need to check if the version in the XML is the same as the one in the file name(NameNorm), in case the dev just clones the file and only updates the version in the XML.

Reply:
I am not sure, what about the entitlements with no version info or doesnt have versioning enabled? Let me add some tests for that scenario

Reply:
I have addressed this in the latest commit